### ISSUE TITLE
Unused functions ignored in /pkg but not in /internal/pkg

### ISSUE THREAD

**Issue by csandanov** [golangci/golangci-lint#546]
Version of golangci-lint: v1.16.0

I've noticed that unused functions ignored under `/pkg` but not under `/internal/pkg`. My guess is linter considers code under `/pkg` exportable, thus less strict rules. However, in some projects  `/pkg` used as `/internal/pkg` (especially in private repos when no pkg will be exported), is there a way to configure linter to use the same rules for `/pkg`?

**Comment by tpounds** [comment]
@csandanov Thanks for the report! Is this still an issue with the latest release versions? Would it be possible to provide some more context or a minimal example that reproduces the behavior?

**Comment by csandanov** [comment]
Yes, the issue is still there with the latest version. I run linter from the root of my project. In one of my packages under `/pkg` I have a function defined that not used anywhere. Linter doesn't report it although it reports unused methods.

**Comment by tpounds** [comment]
@csandanov Can you explain the following statement in a bit more detail?

>  I have a function defined that not used anywhere. Linter doesn't report it although it reports unused methods.

Does this mean it reports unused methods just not the ones you are expecting? Also, it would be incredibly helpful if you could provide a minimal test case that reproduces the issue.

**Comment by csandanov** [comment]
Ok, it seems it's not a bug after all. It reports private unused functions but not public which I guess makes sense since you can't tell whether it's a generic public package or not. I have `main.go` file with the following code.
```go
package main

import "github.com/csandanov/golinttest/pkg/test"

func main() {
	test.UsedFunc()
}
```
And the local package `pkg/test/test.go`:
```
package test

func UsedFunc() {}

func UnusedFunc() {}

func unusedFunc() {}
```

and linter reports:
```go
pkg/test/test.go:7:6: `unusedFunc` is unused (deadcode)
func unusedFunc() {}
     ^
```

but not public `UnusedFunc`. The behavior is the same if I move the package to `/internal/pkg`.

I'm sorry for wasting your time.

**Comment by tpounds** [comment]
@csandanov Makes sense. As you discovered public functions will not be reported as unused since they are technically part of a public API and could be adopted by users of potential libraries.
